services/cd-service/pkg/repository/repository.go

```diff
@@ -2330,6 +2336,321 @@ func (s *State) WriteAllCommitEvents(ctx context.Context, transaction *sql.Tx, d
 	return nil
 }
 
+func (s *State) DBInsertApplicationWithOverview(ctx context.Context, transaction *sql.Tx, appName string, previousEslVersion db.EslVersion, stateChange db.AppStateChange, metaData db.DBAppMetaData) error {
+	h := s.DBHandler
+	err := h.DBInsertApplication(ctx, transaction, appName, previousEslVersion, stateChange, metaData)
+	if err != nil {
+		return err
+	}
+
+	cache, err := h.ReadLatestOverviewCache(ctx, transaction)
+	if err != nil {
+		return err
+	}
+	if cache == nil {
+		logger.FromContext(ctx).Sugar().Warnf("overview was nil, will skip update for app %s", appName)
+		return nil
+	}
+
+	shouldDelete := stateChange == db.AppStateChangeDelete
+	err = s.UpdateTopLevelApp(ctx, transaction, appName, cache, shouldDelete)
+	if err != nil {
+		return err
+	}
+
+	for envGroupIndex := range cache.EnvironmentGroups {
+		envGroup := cache.EnvironmentGroups[envGroupIndex]
+
+		for i := range envGroup.Environments {
+			env := envGroup.Environments[i]
+
+			if shouldDelete {
+				delete(env.Applications, appName)
+			} else {
+				envApp, err := s.UpdateOneAppEnv(ctx, transaction, appName, env.Name, nil)
+				if err != nil {
+					return err
+				}
+
+				if env.Applications == nil {
+					env.Applications = map[string]*api.Environment_Application{}
+				}
+				env.Applications[appName] = envApp
+			}
+		}
+	}
+
+	err = h.WriteOverviewCache(ctx, transaction, cache)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *State) UpdateTopLevelApp(ctx context.Context, transaction *sql.Tx, appName string, result *api.GetOverviewResponse, deleteApp bool) error {
```

Comment:
This function name doesn't indicate that it's changing overview and can be confusing. I think `UpdateTopLevelAppInOverview` would be better`.

Reply:
done